libmscore/measure.cpp

```diff
@@ -2366,7 +2366,7 @@ void Measure::readVoice(XmlReader& e, int staffIdx, bool irregular)
                         }
                   startingBeam = beam;
                   }
-            else if (tag == "Segment")
+            else if (tag == "Segment" && segment)
```

Comment:
Can't this fix lead to this segment's properties being lost at reading a file? `<Segment>` tags seem to be written only once, so if some segment contains no elements in the first staff it would still probably be written in that staff where no relevant elements for it are present: 
https://github.com/musescore/MuseScore/blob/eb94a63ab1ad72532deee34469252fb3efcf29cf/libmscore/scorefile.cpp#L1183-L1184
https://github.com/musescore/MuseScore/blob/eb94a63ab1ad72532deee34469252fb3efcf29cf/libmscore/scorefile.cpp#L1290

Maybe a more correct fix would be to call `segment = getSegment(...)` like it is done for [other elements](https://github.com/musescore/MuseScore/blob/eb94a63ab1ad72532deee34469252fb3efcf29cf/libmscore/measure.cpp#L2131) but it may happen to be unclear how to guess the correct segment type, and we cannot probably ensure that this segment will actually have elements after reading the whole file.

Reply:
Or, as an alternative, we can alter a segment writing code if it turns out that segments cannot be read correctly in such situations with the current way of writing them.

Reply:
`getSegment()` indeed needs a `SegmentType` which we don't know here, so we can't use that.
Modifying the segment writing code... how? We can't read that particular broken segment, so what to write then? There basically is nothing to write, as nothing got read. But yes, there is information getting lost, as outlined in the issue:

     $ diff -u ~/Desktop/Three_Etudes_1_copy[34].mscx
     --- /c/Users/Jojo/Desktop/Three_Etudes_1_copy3.mscx     2019-08-17 18:55:18.000000000 
     +0200
     +++ /c/Users/Jojo/Desktop/Three_Etudes_1_copy4.mscx     2019-08-17 19:16:06.000000000 
     +0200
     @@ -8675,16 +8675,7 @@
                </voice>
              <voice>
                <location>
     -            <fractions>1/2</fractions>
     -            </location>
     -          <location>
     -            <fractions>1/4</fractions>
     -            </location>
     -          <Segment>
     -            <leadingSpace>0.7</leadingSpace>
     -            </Segment>
     -          <location>
     -            <fractions>1/16</fractions>
     +            <fractions>13/16</fractions>
                  </location>
                <StaffText>
                  <offset x="3.10725" y="-1.78571"/>

Reply:
I mean, old files would probably require this fix to prevent crashes, but then it would be good to write new files in a way that allows a file to be read fully correctly. It looks like the best option in case we cannot handle the old way of writing a file well.

Reply:
Ah, I see, you mean to prevent the issue from occuring in the first place in new scores

Reply:
So in `Segment::write()`? That does protect itself from running twice.
Hmm, I guess that <segment> tag needs to get removed along with the chord/rest it belongs to?

Reply:
Question is whether this is worth the effort, considering how rare the use-case/chance is: non-voice 1 rests with leading space on the non-first one getting deleted in a certain order?

Reply:
@mattmcclinch's fix seems simple enough to make it worth the effort...

Reply:
Probably so, if a segment contains some other chord or non-gap rest then its information will not be lost, and if it doesn't then it doesn't make sense anyway. Some issues may be possible if a segment has annotations but no elements or only gaps, but that seems to be true even without this fix (at least for the case of no elements) so should better be fixed separately if it is the case. So yes, I believe this fix indeed can be applied.